backends/gdx-backends-gwt/src/com/badlogic/gdx/backends/gwt/preloader/Preloader.java

```diff
@@ -241,56 +246,44 @@ public boolean isDirectory (String url) {
 		return directories.containsKey(url);
 	}
 
-	private boolean isChild(String path, String url) {
-		return path.startsWith(url) && (path.indexOf('/', url.length() + 1) < 0);
-	}
-
-	public FileHandle[] list (String url) {
-		Array<FileHandle> files = new Array<FileHandle>();
-		for (String path : texts.keys()) {
-			if (isChild(path, url)) {
-				files.add(new GwtFileHandle(this, path, FileType.Internal));
+	public FileHandle[] list (final String url) {
+		return getMatchedAssetFiles(new FilePathFilter() {
+			@SuppressWarnings("synthetic-access")
+			@Override
+			public boolean accept (String path) {
+				return isChild(path, url);
 			}
-		}
-		FileHandle[] list = new FileHandle[files.size];
-		System.arraycopy(files.items, 0, list, 0, list.length);
-		return list;
+		});
 	}
 
-	public FileHandle[] list (String url, FileFilter filter) {
-		Array<FileHandle> files = new Array<FileHandle>();
-		for (String path : texts.keys()) {
-			if (isChild(path, url) && filter.accept(new File(path))) {
-				files.add(new GwtFileHandle(this, path, FileType.Internal));
+	public FileHandle[] list (final String url, final FileFilter filter) {
+		return getMatchedAssetFiles(new FilePathFilter() {
+			@SuppressWarnings("synthetic-access")
```

Comment:
Ok, I'll make the Preloader.isChild() method package-visible, since it being private resulted in the warnings. Those synthetic-access warnings seem to be relatively new, I can't remember any warnings when accessing private methods of the parent class from a nested class before lately. This is even officially supported, it's just that the compiler would auto-generate some protected accessor methods or something...